crates/lair_keystore/src/store_sqlite.rs

```diff
@@ -218,6 +304,168 @@ impl SqlPool {
+    fn get_bidi_ctx_key(&self) -> sodoken::BufReadSized<32> {
+        self.0.lock().ctx_secret.clone()
+    }
+
+    fn list_entries(
+        &self,
+    ) -> BoxFuture<'static, LairResult<Vec<LairEntryInfo>>> {
+        let read = self.read();
+        async move {
+            let mut read = read.await;
+            read.transaction(|txn| {
+                let mut s = txn
+                    .prepare("SELECT data FROM lair_keystore;")
+                    .map_err(one_err::OneErr::new)?;
+                let it = s
+                    .query_map([], |row| {
+                        // go ahead and clone here, so we're not doing the
+                        // decoding while the database is connected...
+                        let data: Vec<u8> = row.get(0)?;
+                        Ok(data.into_boxed_slice())
+                    })
+                    .map_err(one_err::OneErr::new)?;
+                let mut out = Vec::new();
+                for i in it {
+                    let e = LairEntryInner::decode(
+                        &i.map_err(one_err::OneErr::new)?,
+                    )?;
+                    let e = match e {
+                        LairEntryInner::Seed { tag, seed_info, .. } => {
+                            LairEntryInfo::Seed { tag, seed_info }
+                        }
+                        LairEntryInner::DeepLockedSeed {
+                            tag,
+                            seed_info,
+                            ..
+                        } => LairEntryInfo::DeepLockedSeed { tag, seed_info },
+                        LairEntryInner::WkaTlsCert {
+                            tag, cert_info, ..
+                        } => LairEntryInfo::WkaTlsCert { tag, cert_info },
+                        _ => continue,
+                    };
+                    out.push(e);
+                }
+                Ok(out)
+            })
+            .await
+        }
+        .boxed()
+    }
+
+    fn write_entry(
+        &self,
+        entry: LairEntry,
+    ) -> BoxFuture<'static, LairResult<()>> {
+        let this = self.clone();
+        async move {
+            let seed_info = match &*entry {
+                LairEntryInner::Seed { seed_info, .. } => {
+                    Some(seed_info.clone())
+                }
+                LairEntryInner::DeepLockedSeed { seed_info, .. } => {
+                    Some(seed_info.clone())
+                }
+                _ => None,
+            };
+            let bytes = entry.encode()?;
+            let mut write = this.write().await;
+            if let Some(seed_info) = seed_info {
+                write
+                    .transaction(move |txn| {
+                        // if the entry type has seed info
+                        // add the extra columns so we can look up
+                        // the entry by public keys
+                        txn.execute_optional(
+                            r#"INSERT INTO lair_keystore (
```

Comment:
should we have SQL in a separate file like we did with the scheduler?

Reply:
:+1: done